In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
from datetime import datetime

In [34]:
df = pd.read_csv('/content/Unicorn Companies Data (Before Preprocessing).csv')
df.head()

Company Valuation ($B) Date Joined        Country           City  \
0   Bytedance           $140    4/7/2017          China        Beijing   
1      SpaceX         $100.3   12/1/2012  United States      Hawthorne   
2      Stripe            $95   1/23/2014  United States  San Francisco   
3      Klarna          $45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            $42  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

  Founded Year Total Raised Financial Stage Investors Count Deal Terms  \
0         2012       $7.44B             IPO              28          8   
1         2002      $6.874B            None              29         12   
2         2010      $2.901B           Asset              39         12   
3         2005      $3.472B        Acquired              56         13   
4         1991      $4.377B        Acquired              25          5   

  Portfolio Exits  
0               5  
1            None  
2               1  
3               1  
4               2

In [4]:
df.shape

(1037, 13)

In [5]:
df.isnull().sum()

Company              0
Valuation ($B)       0
Date Joined          0
Country              0
City                 0
Industry             0
Select Inverstors    0
Founded Year         0
Total Raised         0
Financial Stage      0
Investors Count      0
Deal Terms           0
Portfolio Exits      0
dtype: int64

In [6]:
df.drop(['Financial Stage', 'Portfolio Exits'], axis = 1, inplace = True)

In [7]:
df.dropna(inplace = True)

In [8]:
df.isnull().sum()

Company              0
Valuation ($B)       0
Date Joined          0
Country              0
City                 0
Industry             0
Select Inverstors    0
Founded Year         0
Total Raised         0
Investors Count      0
Deal Terms           0
dtype: int64

In [9]:
df.dtypes

Company              object
Valuation ($B)       object
Date Joined          object
Country              object
City                 object
Industry             object
Select Inverstors    object
Founded Year         object
Total Raised         object
Investors Count      object
Deal Terms           object
dtype: object

# Object to Numerical [converting every value in Total Raised (billions) to simple numbers]

+ Valuation: $140

+ Total Raised: $7.44B

In [10]:
# bm = []
# t_rows = []
# for rows in df['Total Raised']:
#     bm.append(rows[-1])
#     t_rows.append(float(rows[:-1]))
# df['Total Raised Unit'] = bm
# df['Total Raised Unit'].value_counts()
# OR
df["Total Raised Unit"] = df["Total Raised"].str[-1]

df["Total Raised"] = df["Total Raised"].replace({"K":"", "B":"", "M":"", "None":np.nan}, regex = True)

In [11]:
df.head()

Company Valuation ($B) Date Joined        Country           City  \
0   Bytedance           $140    4/7/2017          China        Beijing   
1      SpaceX         $100.3   12/1/2012  United States      Hawthorne   
2      Stripe            $95   1/23/2014  United States  San Francisco   
3      Klarna          $45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            $42  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

  Founded Year Total Raised Investors Count Deal Terms Total Raised Unit  
0         2012        $7.44              28          8                 B  
1         2002       $6.874              29         12                 B  
2         2010       $2.901              39         12                 B  
3         2005       $3.472              56         13                 B  
4         1991       $4.377              25          5                 B

In [12]:
cols = ["Valuation ($B)", "Total Raised"]
for col in cols:
    df[col] = df[col].str.replace("$", "")

df['Total Raised'] = df['Total Raised'].astype('float')
df['Valuation ($B)'] = df['Valuation ($B)'].astype('float')

df.head()

<ipython-input-12-d1d09f5c35f0>:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df[col] = df[col].str.replace("$", "")


Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0    4/7/2017          China        Beijing   
1      SpaceX           100.3   12/1/2012  United States      Hawthorne   
2      Stripe            95.0   1/23/2014  United States  San Francisco   
3      Klarna            45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            42.0  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

  Founded Year  Total Raised Investors Count Deal Terms Total Raised Unit  
0         2012         7.440              28          8                 B  
1         2002         6.874              29         12                 B  
2         2010         2.901              39         12                 B  
3         2005         3.472              56         13                 B  
4         1991         4.377              25          5                 B

In [13]:
for i, row in df.iterrows():
  if row["Total Raised Unit"] == "B":
    df.loc[i , "Total Raised"] = row["Total Raised"] * 1_000_000_000

  elif row["Total Raised Unit"] == "M":
    df.loc[i, "Total Raised"] = row["Total Raised"] * 1_000_000

  elif row["Total Raised Unit"] == "K":
    df.loc[i, "Total Raised"] = row["Total Raised"] * 1_000

df = df.drop("Total Raised Unit", axis=1)

df.head()

Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0    4/7/2017          China        Beijing   
1      SpaceX           100.3   12/1/2012  United States      Hawthorne   
2      Stripe            95.0   1/23/2014  United States  San Francisco   
3      Klarna            45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            42.0  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

  Founded Year  Total Raised Investors Count Deal Terms  
0         2012  7.440000e+09              28          8  
1         2002  6.874000e+09              29         12  
2         2010  2.901000e+09              39         12  
3         2005  3.472000e+09              56         13  
4         1991  4.377000e+09              25          5

In [23]:
df=df.drop(df[df["Founded Year"]=="None"].index)
df=df.drop(df[df["Total Raised"]=="None"].index)
df=df.drop(df[df["Investors Count"]=="None"].index)
df=df.drop(df[df["Select Inverstors"]=="None"].index)
df=df.drop(df[df["Deal Terms"]=="None"].index)

df.head()

Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0    4/7/2017          China        Beijing   
1      SpaceX           100.3   12/1/2012  United States      Hawthorne   
2      Stripe            95.0   1/23/2014  United States  San Francisco   
3      Klarna            45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            42.0  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial Intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

   Founded Year  Total Raised  Investors Count Deal Terms  
0          2012  7.440000e+09               28          8  
1          2002  6.874000e+09               29         12  
2          2010  2.901000e+09               39         12  
3          2005  3.472000e+09               56         13  
4          1991  4.377000e+09               25          5

In [24]:
df["Industry"]=df["Industry"].str.replace("Artificial intelligence","Artificial Intelligence")
df["Industry"]=df["Industry"].str.replace("Finttech","Fintech")

df["Valuation ($B)"]=df["Valuation ($B)"].astype(float)
df["Investors Count"]=df["Investors Count"].astype(int)
df['Founded Year'] = df['Founded Year'].astype('int')
df['Deal Terms'] = df['Deal Terms'].astype('int')


df.head()

Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0    4/7/2017          China        Beijing   
1      SpaceX           100.3   12/1/2012  United States      Hawthorne   
2      Stripe            95.0   1/23/2014  United States  San Francisco   
3      Klarna            45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            42.0  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial Intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

   Founded Year  Total Raised  Investors Count  Deal Terms  
0          2012  7.440000e+09               28           8  
1          2002  6.874000e+09               29          12  
2          2010  2.901000e+09               39          12  
3          2005  3.472000e+09               56          13  
4          1991  4.377000e+09               25           5

In [25]:
df.dtypes

Company               object
Valuation ($B)       float64
Date Joined           object
Country               object
City                  object
Industry              object
Select Inverstors     object
Founded Year           int64
Total Raised         float64
Investors Count        int64
Deal Terms             int64
dtype: object

# Converting Date into DateTime type

In [26]:
df[['Date','Month','Year']]=df['Date Joined'].str.split('/',expand=True)
df.head()

Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0    4/7/2017          China        Beijing   
1      SpaceX           100.3   12/1/2012  United States      Hawthorne   
2      Stripe            95.0   1/23/2014  United States  San Francisco   
3      Klarna            45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            42.0  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial Intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

   Founded Year  Total Raised  Investors Count  Deal Terms Date Month  Year  
0          2012  7.440000e+09               28           8    4     7  2017  
1          2002  6.874000e+09               29          12   12     1  2012  
2          2010  2.901000e+09               39          12    1    23  2014  
3          2005  3.472000e+09               56          13   12    12  2011  
4          1991  4.377000e+09               25           5   10    26  2018

In [27]:
df["Date Joined"]=pd.to_datetime(df["Date Joined"])

df["Date"]=df["Date"].astype(int)
df["Month"]=df["Month"].astype(int)
df["Year"]=df["Year"].astype(int)
df["Founded Year"]=df["Founded Year"].astype(int)

In [28]:
df.dtypes

Company                      object
Valuation ($B)              float64
Date Joined          datetime64[ns]
Country                      object
City                         object
Industry                     object
Select Inverstors            object
Founded Year                  int64
Total Raised                float64
Investors Count               int64
Deal Terms                    int64
Date                          int64
Month                         int64
Year                          int64
dtype: object

## Years taken to become Unicorn

In [29]:
df['Years Taken to become Unicorn'] = df['Year'] - df['Founded Year']
df['Years Taken to become Unicorn']=df['Years Taken to become Unicorn'].astype(int)
df.head()

Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0  2017-04-07          China        Beijing   
1      SpaceX           100.3  2012-12-01  United States      Hawthorne   
2      Stripe            95.0  2014-01-23  United States  San Francisco   
3      Klarna            45.6  2011-12-12         Sweden      Stockholm   
4  Epic Games            42.0  2018-10-26  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial Intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

   Founded Year  Total Raised  Investors Count  Deal Terms  Date  Month  Year  \
0          2012  7.440000e+09               28           8     4      7  2017   
1          2002  6.874000e+09               29          12    12      1  2012   
2          2010  2.901000e+09               39          12     1     23  2014   
3          2005  3.472000e+09               56          13    12     12  2011   
4          1991  4.377000e+09               25           5    10     26  2018   

   Years Taken to become Unicorn  
0                              5  
1                             10  
2                              4  
3                              6  
4                             27

In [35]:
df.to_csv('Unicorn Companies Data (After Preprocessing).csv', index=False)

In [31]:
df.describe()

Valuation ($B)  Founded Year  Total Raised  Investors Count  \
count      958.000000    958.000000  9.460000e+02       958.000000   
mean         3.348236   2012.672234  5.891447e+08        14.744259   
std          7.521574      5.957976  7.539080e+08         9.741934   
min          1.000000   1919.000000  5.000000e+06         1.000000   
25%          1.100000   2011.000000  2.359025e+08         8.000000   
50%          1.600000   2014.000000  3.832150e+08        13.000000   
75%          3.137500   2016.000000  6.454500e+08        19.000000   
max        140.000000   2021.000000  7.440000e+09        91.000000   

       Deal Terms        Date       Month         Year  \
count  958.000000  958.000000  958.000000   958.000000   
mean     3.128392    6.448852   15.921712  2019.842380   
std      2.172115    3.526137    8.548291     1.969251   
min      1.000000    1.000000    1.000000  2011.000000   
25%      1.000000    3.000000    9.000000  2019.000000   
50%      3.000000    6.000000   16.000000  2021.000000   
75%      4.000000   10.000000   23.000000  2021.000000   
max     19.000000   12.000000   31.000000  2022.000000   

       Years Taken to become Unicorn  
count                     958.000000  
mean                        7.170146  
std                         5.543338  
min                         0.000000  
25%                         4.000000  
50%                         6.000000  
75%                         9.000000  
max                        98.000000

In [32]:
df.nunique()

Company                          956
Valuation ($B)                   187
Date Joined                      591
Country                           43
City                             241
Industry                          15
Select Inverstors                946
Founded Year                      36
Total Raised                     863
Investors Count                   52
Deal Terms                        15
Date                              12
Month                             31
Year                              12
Years Taken to become Unicorn     30
dtype: int64

# COUNTRY WISE ANALYSIS

## Top Countries wrt the Total Valuation and Number of Unicorns

In [ ]:
temp = df.groupby(by=['Country']).agg({'Company':['count'],'Valuation ($B)': ['sum']})
temp = temp.sort_values([('Company', 'count')], ascending = False)
temp.head(10)

In [ ]:
fig=px.bar(x=temp[('Valuation ($B)','sum')],y=temp.index,title="Top Countries wrt Total Valuation and the Number of Unicorns",text_auto=True,color=temp[('Company', 'count')], labels={
                     "y": "Country",
                     "x": "Total Valuation($B)",
                     "color":"Number of Unicorns"})
fig.show()

## Average Years taken to become Unicorn wrt Country

In [ ]:
temp = df.groupby(by=['Country']).agg({'Years Taken to become Unicorn':['mean']})
temp = temp.sort_values([('Years Taken to become Unicorn', 'mean')], ascending=False)
temp.head(10)

In [ ]:
fig=px.scatter(y=temp[('Years Taken to become Unicorn','mean')],x=temp.index,size=temp[('Years Taken to become Unicorn', 'mean')],size_max=30,color=temp.index,title="Average years taken to become Unicorn wrt Country",labels={"y":"Years Taken","x":"Country", "color": "Country"})
fig.show()

In [ ]:
fig=px.line(y=temp[('Years Taken to become Unicorn','mean')],x=temp.index,labels={
                     "x": "Country",
                     "y": "Average Years Taken"},title="Average Years taken to become Unicorn wrt Country")
fig.update_traces(line_color="blue")
fig.show()

## Top Countries wrt Total Money raised

In [ ]:
temp=df.groupby(by=['Country']).agg({'Total Raised':['sum']})
temp=temp.sort_values([('Total Raised', 'sum')], ascending=False)[:6]
temp

In [ ]:
fig=px.line(y=temp[('Total Raised', 'sum')],x=temp.index,title="Top Countries wrt Total Money raised",labels={"y":"Total Raised($B)","x":"Country"})
fig.update_traces(line_color="blue")
fig.show()

# CITY WISE ANALYSIS

## Top Cities wrt the Total Valuation and Number of Unicorns

In [ ]:
temp=df.groupby(by=['City']).agg({'Company':['count'],'Valuation ($B)': ['sum']})
temp=temp.sort_values([('Valuation ($B)', 'sum')], ascending=False)[:7]
temp

In [ ]:
fig=px.bar(x=temp[('Valuation ($B)','sum')],y=temp.index,title="Top Cities wrt Total Valuation and the Number of Unicorns",text_auto=True,color=temp[('Company', 'count')], labels={
                     "y": "City",
                     "x": "Total Valuation($B)",
                     "color":"Number of Unicorns"})
fig.show()

## Years taken to become Unicorn wrt Country and City

In [ ]:
fig=px.scatter(df,x='Years Taken to become Unicorn',y='Country',hover_data=['City'],color='City',title="Years taken to become Unicorn wrt Country and City")
fig.show()

## Average Years taken to become Unicorn wrt City

In [ ]:
temp=df.groupby(by=['City']).agg({'Years Taken to become Unicorn':['mean']})
temp=temp.sort_values([('Years Taken to become Unicorn', 'mean')], ascending=False)[115:135]
temp

Years Taken to become Unicorn
                                       mean
City                                       
Croix                              7.000000
Gurgaon                            6.875000
San Diego                          6.833333
Austin                             6.800000
Gurugram                           6.750000
Scottsdale                         6.666667
Bellevue                           6.600000
New York                           6.577778
Shenzhen                           6.529412
Burlingame                         6.500000
Culver City                        6.500000
Irvine                             6.500000
Curitiba                           6.500000
San Francisco                      6.469697
Vancouver                          6.400000
Hangzhou                           6.384615
Seattle                            6.285714
Miami                              6.200000
Sunnyvale                          6.166667
Atlanta                            6.142857

In [ ]:
fig=px.scatter(y=temp[('Years Taken to become Unicorn','mean')],x=temp.index,size=temp[('Years Taken to become Unicorn', 'mean')],color=temp[('Years Taken to become Unicorn', 'mean')],title="Average years taken to become Unicorn wrt City",labels={"y":"Years Taken","x":"City", "color": "Years Taken"})
fig.show()

## Top Cities wrt Total Money raised

In [ ]:
temp=df.groupby(by=['City']).agg({'Total Raised':['sum']})
temp=temp.sort_values([('Total Raised', 'sum')], ascending=False)[:9]
temp

Total Raised
                        sum
City                       
San Francisco  6.760940e+10
Beijing        5.133098e+10
New York       3.570104e+10
Shanghai       2.555420e+10
London         2.413281e+10
Bengaluru      2.229692e+10
Berlin         1.323375e+10
Stockholm      1.134985e+10
Paris          1.011609e+10

In [ ]:
fig=px.line(y=temp[('Total Raised', 'sum')],x=temp.index,title="Top Cities wrt Total Money raised",labels={"y":"Total Raised($B)","x":"City"})
fig.update_traces(line_color="blue")
fig.show()

# INDUSTRY WISE ANALYSIS

## Top Industries wrt the Total Valuation and Number of Unicorns

In [ ]:
temp=df.groupby(by=['Industry']).agg({'Company':['count'],'Valuation ($B)': ['sum','mean']})
temp=temp.sort_values([('Company', 'count')], ascending=False)[:6]
temp

Company Valuation ($B)          
                                  count            sum      mean
Industry                                                        
Fintech                             188         747.16  3.974255
Internet software & services        179         529.25  2.956704
E-commerce & direct-to-consumer     100         311.85  3.118500
Artificial intelligence              65         286.68  4.410462
Health                               65         174.15  2.679231
Other                                52         259.20  4.984615

In [ ]:
fig=px.bar(x=temp[('Valuation ($B)','sum')],y=temp.index,title="Top Industries wrt Total Valuation and the Number of Unicorns",text_auto=True,color=temp[('Company', 'count')],labels={
                     "y": "Industry",
                     "x": "Total Valuation($B)",
                     "color":"Number of Unicorns"})

fig.show()

## Top Industries wrt Average Valuation

In [ ]:
temp=temp.sort_values([('Valuation ($B)', 'mean')], ascending=False)[:6]
temp

Company Valuation ($B)          
                                  count            sum      mean
Industry                                                        
Other                                52         259.20  4.984615
Artificial intelligence              65         286.68  4.410462
Fintech                             188         747.16  3.974255
E-commerce & direct-to-consumer     100         311.85  3.118500
Internet software & services        179         529.25  2.956704
Health                               65         174.15  2.679231

In [ ]:
fig=px.bar(x=temp[('Valuation ($B)', 'mean')],y=temp.index,title="Top Industries wrt Average Valuation",text_auto=True,labels={
                     "y": "Industry", "x": "Average Valuation($B)",
                   })

fig.show()

## Years taken to become Unicorn wrt Country and Industry

In [ ]:
fig=px.scatter(df,x='Years Taken to become Unicorn',y='Country',hover_data=['Industry'],color='Industry',title="Years taken to become Unicorn wrt Country and Industry")
fig.show()

## Average Years taken to become Unicorn wrt Industry

In [ ]:
temp=df.groupby(by=['Industry']).agg({'Years Taken to become Unicorn':['mean']})
temp=temp.sort_values([('Years Taken to become Unicorn', 'mean')], ascending=False)[2:15]
temp

Years Taken to become Unicorn
                                                             mean
Industry                                                         
Internet software & services                             8.011173
Edtech                                                   7.888889
Data management & analytics                              7.641026
Consumer & retail                                        7.217391
Cybersecurity                                            6.937500
Supply chain, logistics, & delivery                      6.865385
Fintech                                                  6.819149
Artificial intelligence                                  6.753846
Travel                                                   6.571429
E-commerce & direct-to-consumer                          6.340000
Mobile & telecommunications                              6.103448
Hardware                                                 5.687500
Artificial Intelligence                                  5.142857

In [ ]:
fig=px.scatter(y=temp[('Years Taken to become Unicorn','mean')],x=temp.index,size=temp[('Years Taken to become Unicorn', 'mean')],size_max=30,color=temp.index,title="Average years taken to become Unicorn wrt Industry",labels={"y":"Years Taken","x":"Industry", "color": "Industry"})
fig.show()

## Top Industries wrt Total Money raised

In [ ]:
temp=df.groupby(by=['Industry']).agg({'Total Raised':['sum']})
temp=temp.sort_values([('Total Raised', 'sum')], ascending=False)[:10]
temp

Total Raised
                                              sum
Industry                                         
Fintech                              9.481014e+10
E-commerce & direct-to-consumer      7.097430e+10
Internet software & services         7.026770e+10
Other                                4.237474e+10
Supply chain, logistics, & delivery  4.079252e+10
Artificial intelligence              3.805034e+10
Auto & transportation                3.390202e+10
Health                               3.332803e+10
Hardware                             2.437135e+10
Cybersecurity                        2.222740e+10

In [ ]:
fig=px.line(y=temp[('Total Raised', 'sum')],x=temp.index,title="Top Industries wrt Total Money raised",labels={"y":"Total Raised($B)","x":"Industry"})
fig.update_traces(line_color="blue")
fig.show()

# INVESTORS,TIME and OTHER ANALYSIS

## Number of Unicorns wrt Year Joined

In [ ]:
temp=df.groupby(by=['Year']).agg({'Company':['count']})
temp=temp.sort_values([('Company', 'count')], ascending=False)[:5]
temp

Company
       count
Year        
2021     466
2020     100
2018      94
2019      93
2022      81

In [ ]:
fig=px.pie(values=temp[('Company', 'count')],names=temp.index,title="Years wrt the highest number of Companies that became Unicorns")
fig.update_traces(textinfo='value',textfont_size=20)
fig.show()

## Founded Years wrt number of Companies that became Unicorn

In [ ]:
temp=df.groupby(by=['Founded Year']).agg({'Company':['count']})
temp=temp.sort_values([('Company','count')], ascending=False)[:10]
temp

Company
               count
Founded Year        
2015             137
2014             100
2016              97
2012              85
2013              82
2011              73
2017              64
2018              51
2010              34
2009              33

In [ ]:
fig=px.bar(y=temp[('Company','count')],x=temp.index,title="Founded Years wrt number of Companies that became Unicorn",text_auto=True,labels={"x":"Founded Years","y":"Number of Unicorns"})
fig.show()

## Top Investors wrt Number of Unicorns they invested in

In [ ]:
investors = []
for i, row in df.iterrows():

        investors += row["Select Inverstors"].split(', ')
investors = pd.Series(investors).value_counts()[:10]
investors.sort_values(ascending=False, inplace=True)
investors

Accel                          54
Tiger Global Management        48
Andreessen Horowitz            41
Sequoia Capital China          39
Insight Partners               39
Sequoia Capital                36
Lightspeed Venture Partners    32
SoftBank Group                 29
General Catalyst               29
Index Ventures                 26
Name: count, dtype: int64

In [ ]:
px.bar (investors.index,investors.values)
fig=px.bar(y=investors.values,x=investors.index,title="Top Investors wrt the Number of Unicorns they invested in",text_auto=True,labels={"x":"Investors","y":"Number of Unicorns"})
fig.show()

## Total Raised wrt Valuation

In [ ]:
temp=df.sort_values("Valuation ($B)", ascending=False)[:13]
fig=px.scatter(x=temp['Valuation ($B)'], y=temp['Total Raised'],size=temp['Years Taken to become Unicorn'] ,hover_name=temp['Company'],color=temp['Company'],size_max=70,title="Total Raised wrt the Valuation of Top Unicorns(Size of the bubble: Years taken to become Unicorn)",labels={"x":"Valuation($B)","y":"Total Raised($B)"})
fig.show()